pkg/api/v1/clients/kube/crd/crd.go

```diff
@@ -74,17 +74,29 @@ func (d Crd) Register(apiexts apiexts.Interface) error {
 }
 
 func (d Crd) KubeResource(resource resources.InputResource) *v1.Resource {
+
+	// Handle spec
 	data, err := protoutils.MarshalMap(resource)
 	if err != nil {
 		panic(fmt.Sprintf("internal error: failed to marshal resource to map: %v", err))
 	}
+
 	delete(data, "metadata")
 	delete(data, "status")
 	spec := v1.Spec(data)
+
```

Comment:
Can't you potentially just do the same here with the status, rather than marshalling again?
```
status, ok := data["status"]
if ok {
    status := v1.Status(status)
}
```

Reply:
Potentially, but I think we should use the `GetStatus()` function on the input `InputResource` as the source of truth for the status.